src/erldns_txt.erl

```diff
@@ -77,15 +78,91 @@ split(Data, Parts) ->
 
 -ifdef(TEST).
 
+
+%% Known failure cases:
+%% \
+%% "
+%% "\"
 parse_test() ->
-  ?assertEqual(parse(""), []),
-  ?assertEqual(parse("test"), [[<<"test">>]]),
-  ?assertEqual(parse(lists:duplicate(270, "x")), [[list_to_binary(lists:duplicate(255, "x")), list_to_binary(lists:duplicate(15, "x"))]]),
-  ?assertEqual(parse(<<"test">>), [[<<"test">>]]),
-  ?assertEqual(parse("\"test\" \"test\""), [[<<"test">>], [<<"test">>]]),
-  ?assertEqual(parse("\\"), [[<<"\\">>]]),
-  ?assertEqual(parse("test\\;"), [[<<"test\\;">>]]),
-  ?assertEqual(parse("test\\"), [[<<"test\\">>]]).
+  ?assertEqual([], parse("")),
+  ?assertEqual([[<<"test">>]], parse("test")),
+  ?assertEqual([[list_to_binary(lists:duplicate(255, "x")), list_to_binary(lists:duplicate(15, "x"))]], parse(lists:duplicate(270, "x"))),
+  ?assertEqual([[<<"test">>]], parse(<<"test">>)),
+  ?assertEqual([[<<"test">>], [<<"test">>]], parse("\"test\" \"test\"")),
+  ?assertEqual([[<<"\\">>]], parse("\\")),
+  ?assertEqual([[<<"test\\;">>]], parse("test\\;")),
+  ?assertEqual([[<<"test\\">>]], parse("test\\")).
 %?assertEqual(parse("\"test\"\""), [[<<"test\"">>]]).
 
+proper_test_() ->
```

Comment:
Added some model checks. We can get rid of these, or break them out into a separate commit, but it's was a really nice to have while troubleshooting this module.